From ab4b93abbf08e172825ce13daed4e586b49ec212 Mon Sep 17 00:00:00 2001
From: Rich Megginson <rmeggins@redhat.com>
Date: Fri, 28 Jun 2013 11:37:05 -0600
Subject: [PATCH 78/99] Ticket #47409 - allow setting db deadlock rejection
 policy

https://fedorahosted.org/389/ticket/47409
Reviewed by: nhosoi (Thanks!)
Branch: master
Fix Description: Forgot to add flags parameter to last commit.
Platforms tested: RHEL6 x86_64
Flag Day: no
Doc impact: yes - document new config parameter
(cherry picked from commit 67cbb2eac1300ae57acf75f4c3299103046a3323)
(cherry picked from commit ceaa9ac5267c04c23e3fb8e366e17295ffd6b93c)
(cherry picked from commit e535ff5a45601c4279a14e92eaf1d68729866d38)
(cherry picked from commit 6c1a71310292e5992aa1fcc8c203a6b6d595ebe3)
---
 ldap/servers/slapd/back-ldbm/dblayer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/servers/slapd/back-ldbm/dblayer.c b/ldap/servers/slapd/back-ldbm/dblayer.c
index 85e4a18..8696834 100644
--- a/ldap/servers/slapd/back-ldbm/dblayer.c
+++ b/ldap/servers/slapd/back-ldbm/dblayer.c
@@ -4227,7 +4227,7 @@ static int deadlock_threadmain(void *param)
             if (dblayer_db_uses_locking(db_env) && (deadlock_policy > DB_LOCK_NORUN)) {
                 int rejected = 0;
 
-                if ((rval = LOCK_DETECT(db_env, 0, deadlock_policy, &rejected)) != 0) {
+                if ((rval = LOCK_DETECT(db_env, flags, deadlock_policy, &rejected)) != 0) {
                     LDAPDebug(LDAP_DEBUG_ANY,
                               "Serious Error---Failed in deadlock detect (aborted at 0x%x), err=%d (%s)\n",
                               rejected, rval, dblayer_strerror(rval));
-- 
1.8.1.4

